src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -889,9 +889,12 @@ private void WorkspaceAdded(WorkspaceModel item)
             var newVm = new WorkspaceViewModel(item, this);
             if (item is HomeWorkspaceModel)
             {
+                (item as HomeWorkspaceModel).DynamicRunEnabled = DynamicRunEnabled;
```

Comment:
Although ``item`` is not the same as ``HomeSpace`` here, but your point is valid. Here we take the assumption that newly added home workspace model is *after* the current home workspace model, this assumption is brittle. I'll do the change. Thanks!